src/pages/blog/neon.md

```diff
@@ -0,0 +1,495 @@
+
+```rb
+class NotesController < ApplicationController
+    def all
+        @notes = current_user.database do
+            Note.all.pluck(:id, :title)
+        end
+    end
+end
+```
+
+Now we will update the HTML to make this a basic listing:
+
+```html
+<h1>Notes</h1>
+<p>
+    You have <%= pluralize @notes.length, 'note' %>.
+</p>
+
+<ul>
+    <% @notes.each do |note| %>
+        <li>
+            <a href="/notes/<%= note[0] %>">
+                <%= note[1] %>
+            </a>
+        </li>
+    <% end %>
+</ul>
+
+<p>
+    <a href="/notes/new">
+        New Note
+    </a>
+</p>
+```
+
+Now we will make a simple new notes page. Write this to `app/views/notes/new.html.erb`:
+
+```html
+<h1>New Note</h1>
+
+<% if @note.present? %>
+    <ul>
+        <% @note.errors.full_messages.each do |message| %>
+            <li><%= message %></li>
+        <% end %>
+    </ul>
+<% end %>
+
+<%= form_with do |form| %>
+    <%= form.label :title, 'Title' %>
+    <%= form.text_field :title %>
+    <div style="display: block; margin-top: 1rem; margin-bottom: 1rem">
+        <%= form.label :body, 'Body' %>
+        <%= form.text_area :body, size: '70x5' %>
+    </div>
+    <%= form.submit %>
+<% end %>
+```
+
+We can now add this to `config/routes.rb`:
+
+```rb
+# Add the new form handler.
+get "notes/new", to: "notes#new"
+post "notes/new", to: "notes#create"
+get "notes/:id", to: "notes#show"
+```
+
+And add the `create` method to the controller:
+
+```rb
+    def create
+        @note = current_user.database do
+            Note.create(params.permit(:title, :body))
+        end
+        return render :new, status: :bad_request if @note.errors.any?
+        redirect_to "/notes/#{@note.id}"
+    end
+```
+
+And we will want to add constraints to our Note model:
+
+```rb
+class Note < ApplicationRecord
+    validates :title, presence: true, length: { minimum: 2 }
+    validates :body, presence: true, length: { minimum: 2 }
+end
+```
+
+Next, we'll add the `show` method to the controller:
+
+```rb
+    def show
+        @note = current_user.database do
+            Note.find params[:id]
+        end
+    end
+```
+
+And then make `app/views/notes/show.html.erb`:
+
+```html
+<h1><%= @note.title %></h1>
+
+<p><pre><%= @note.body %></pre></p>
+
+<p>
+    <a href="/">Show all notes</a>
+</p>
+```
+
+You now have a basic multi-tenant notes application using Ruby on Rails!
+
+## Conclusion
+
+In conclusion, whilst this method of database storage still has its compromises which you need to consider, this shows a new way of storing data that offers different data protection, data integrity, and potentially cost benefits since each user can be individually scaled up and down without needing to worry about noisy neighbours.
```

Comment:
```suggestion
In conclusion, whilst this method of database storage still has its tradeoffs which you need to consider, this shows a new way of storing data that offers different data protection, data integrity, and potentially cost benefits since each user can be individually scaled up and down without needing to worry about noisy neighbours.
```